### PR TITLE
Bugfix/iat 3205

### PR DESCRIPTION
**Pull request by millerts**

### CODE REVIEW

File: application.yml [hunk 1]
```diff
@@ -62,9 +62,9 @@ migration:
   numberOfThreads: 5
   publishReportEnabled: true
   syncFromGitlabEnabled: false
-  copySourceToTargetEnabled: true
+  copySourceToTargetEnabled: false
   deleteItemsEnabled: false
-  itemFileLocation: "${TIMS_COPY_ITEM_FILE_LOCATION}"
+  itemFileLocation: ${tims.copy.item.file.location}
```
Comment (millerts) [inline application.yml]: This was my attempt to change the item file location.  Unfortunately, the way the copy tool works makes this difficult.  I'll need to play around with it more.

File: src/main/java/org/opentestsystem/ap/migration/migration/Migration3104.java [hunk 2]
```diff
@@ -25,26 +19,21 @@
...
-import java.util.Set;
-import java.util.function.Function;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
 
 import static java.lang.String.format;
 import static org.opentestsystem.ap.common.model.ModelConstants.ItemType.TYPE_TUT;
 
-@Slf4j
-@Component
```
Comment (millerts) [inline src/main/java/org/opentestsystem/ap/migration/migration/Migration3104.java]: Moved below the JavaDoc so it doesn't flag it as a warning.

File: src/main/java/org/opentestsystem/ap/migration/migration/Migration3104.java [hunk 3]
```diff
@@ -25,26 +19,21 @@
...
 import java.util.Collection;
 import java.util.Collections;
-import java.util.HashSet;
 import java.util.List;
-import java.util.Objects;
 import java.util.Optional;
-import java.util.Set;
-import java.util.function.Function;
-import java.util.stream.Collectors;
-import java.util.stream.Stream;
```
Comment (millerts) [inline src/main/java/org/opentestsystem/ap/migration/migration/Migration3104.java]: Not sure the original class but there must have been quite a few unused imports.  I only changed the code blocks in green.

**Comment by millerts** [comment]
Note that this is merging the bugfix branch into the hotfix branch.  This is intended.

File: src/main/java/org/opentestsystem/ap/migration/migration/Migration3104.java [hunk 4]
```diff
@@ -194,4 +189,14 @@ private boolean hasFile(List<Attachment> attachments, String fileName) {
...
     }
+
+    private Optional<String> findSourceFileName(File glossaryDirectory, String fileName) {
+        String[] glossaryFiles = glossaryDirectory.list();
+
+        if(glossaryFiles == null) {
+            return Optional.empty();
+        }
+
+        return Arrays.stream(glossaryFiles).filter(s -> s.equalsIgnoreCase(fileName)).findFirst();
```
Comment (bdrainer) [inline src/main/java/org/opentestsystem/ap/migration/migration/Migration3104.java]: The filter is ignoring casing and that makes sense.  Is there anywhere where we are lower casing the file name so we store it in TIMS all lower cased?   Maybe that doesn't matter.

Comment (millerts) [reply src/main/java/org/opentestsystem/ap/migration/migration/Migration3104.java]: It is hard to see but that is being done in the parent method.  Roughly on this line https://github.com/SmarterApp/AP_ItemMigration/pull/184/files#diff-b581edd821a12d342db690af848ba753R152.  The m4aFileName is created from the ogg file name which is lowercased.  This bug fixes the issue with finding the source file.  The destination file was always lower-cased.

**Pull request merged**